### PR TITLE
feat: add mint components

### PR DESCRIPTION
**Pull request by alessey**
**What changed? Why?**
Adding Mint components

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/nft/components/mint/NFTCreator.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      <Identity
+        className="space-x-2 px-0"
+        address={creatorAddress}
+        schemaId={schemaId}
+      >
+        <Avatar className="h-5 w-5" />
+        <Name>
+          <Badge />
+        </Name>
+      </Identity>
```
Comment (0xAlec) [inline src/nft/components/mint/NFTCreator.tsx:29]: love the composability here

File: src/nft/components/mint/NFTMintButton.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    return null;
+  }
+
+  if (!address) {
+    return (
+      <div className={cn('py-2', className)}>
+        <ConnectWallet className="w-full" />
+      </div>
+    );
+  }
```
Comment (0xAlec) [inline src/nft/components/mint/NFTMintButton.tsx:72]: we should align our components to **_always_** render, and only prompt for wallet connection in response to an action (e.g. minting the NFT)

Comment (abcrane123) [reply src/nft/components/mint/NFTMintButton.tsx:72]: agreed you can see my implementation with SwapButton here https://github.com/coinbase/onchainkit/blob/main/src/swap/components/SwapButton.tsx

Comment (alessey) [reply src/nft/components/mint/NFTMintButton.tsx:72]: I'm missing the difference

Comment (abcrane123) [reply src/nft/components/mint/NFTMintButton.tsx:72]: this is the same implementation as our swap button https://github.com/coinbase/onchainkit/blob/main/src/swap/components/SwapButton.tsx

File: src/nft/components/mint/NFTAssetCost.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { multiplyFloats } from '../../../internal/utils/multiplyFloats';
+import { cn, text } from '../../../styles/theme';
+import { formatAmount } from '../../../token/utils/formatAmount';
+import { useNFTContext } from '../NFTProvider';
+
+type NFTAssetCostReact = {
+  className?: string;
+};
```
Comment (cpcramer) [inline src/nft/components/mint/NFTAssetCost.tsx:8]: Should these types be in a general `types.ts` file?

File: src/nft/components/mint/NFTMintButton.tsx (hunk 2 above)
Comment (abcrane123) [reply src/nft/components/mint/NFTMintButton.tsx:72]: are you suggesting we change the `onClick` handler to be connect wallet if address isn't connected ?

File: src/nft/components/mint/NFTTotalCost.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            })}
+          </div>
+        </div>
+      </div>
+    );
+  }, [mintFee, price, quantity]);
+
+  if (!price?.amount || !price?.currency || !price?.amountUSD) {
+    return null;
+  }
```
Comment (abcrane123) [inline src/nft/components/mint/NFTTotalCost.tsx:88]: is there ever a case where an NFT can be free ?

Comment (alessey) [reply src/nft/components/mint/NFTTotalCost.tsx:88]: yea, the NFTAssetCost component handles that use case, for this total cost component I wasn't showing as we already handle Free elsewhere, but can discuss and revisit.

**Review by 0xAlec — APPROVED** [review]

**Review by cpcramer — APPROVED** [review]

**Pull request merged**